aii-opennebula/src/main/perl/opennebula.pm

```diff
@@ -413,11 +423,17 @@ sub remove_vn_ars
 # Update VN AR leases
 sub update_vn_ars
 {
-    my ($self, $one, $vnet, $ardata, $ar) = @_;
-    my $arid;
+    my ($self, $one, $arinfo, $vnet, $ardata, $ar) = @_;
+    my ($arid, $modtpl);
     # Update the AR info
-    $main::this_app->debug(1, "AR template to update from $vnet: ", $ardata->{ar});
-    $arid = $ar->updatear($ardata->{ar});
+    $arid = $self->detect_vn_ar_quattor($arinfo);
+    if (defined($arid)) {
```

Comment:
@stdweird `ardata` comes from our `$config` template. It includes the info about the AR, IPs, Size etc. This info is used to match an AR within the VNET. The problem is that the RPC `one.vn.update_ar` method does not uses the AR_ID value as argument  :disappointed:, only the AR template and VNET ID... so I have to implement this ugly workaround to fetch and insert the  AR ID in the current template... 

http://docs.opennebula.org/5.0/integration/system_interfaces/api.html#actions-for-virtual-network-management